### PR TITLE
Index and search span status

### PR DESCRIPTION
**Pull request by guilload**
### Description
- index span status
- modify Jaeger query builder to allow `error=true/false` queries

### How was this PR tested?
- added unit tests
- updated existing unit tests

### CODE REVIEW

File: quickwit/quickwit-jaeger/src/integration_tests.rs [hunk 1]
```diff
@@ -226,7 +230,47 @@ async fn test_otel_jaeger_integration() {
...
+            tags: HashMap::from([("error".to_string(), "true".to_string())]),
+            start_time_min: None,
+            start_time_max: None,
+            duration_min: None,
+            duration_max: None,
+            num_traces: 10,
+        };
+        let find_trace_ids_request = FindTraceIDsRequest { query: Some(query) };
+        let find_trace_ids_response = jaeger_service
+            .find_trace_i_ds(tonic::Request::new(find_trace_ids_request))
```
Comment (fmassot) [inline quickwit/quickwit-jaeger/src/integration_tests.rs:248]: is `find_trace_i_ds` the right name? it seems like `find_trace_ids` is.

Comment (guilload) [reply quickwit/quickwit-jaeger/src/integration_tests.rs:248]: codegen

**Review by fmassot — APPROVED** [review]
Approved. Though I think we need to test that in Jaeger UI, is that right?

**Comment by guilload** [comment]
Seems to be working. I generated some `error!` spans in some of our traces and the queries `error:true/false` are working.

**Pull request merged**